agent/config/config.go

```diff
@@ -475,6 +476,13 @@ type PolicyEval struct {
 	Workers map[string]int `hcl:"workers,optional"`
 }
 
+// PolicySource is an individual configured policy source.
+type PolicySource struct {
+	Name       string `hcl:"name,label"`
+	EnabledPtr *bool  `hcl:"enabled,optional"`
+	Enabled    bool
```

Comment:
I am curious about the reasoning for using a pointer along with a non-pointer for the value? Is this so we don't have to dereference every time when reading the value out? If so, could we add a comment to clarify this.

Reply:
I used a pointer so that an empty block (like `source "nomad" {}`) would have `enabled` set to `true`. 

I think it wouldn't be possible to distinguish between an explicit `enabled = false` and a missing `enabled` otherwise? 🤔

Reply:
Yeah I understand the use of the pointer, just curious of the reason for using it alongside a non-pointer rather than referencing the pointer field.

Reply:
Ahh got it. Yeah, that makes sense. I think I was just copy and pasting code 😅